cogs/tmpc.py

```diff
@@ -434,24 +478,90 @@ async def nomod(self, ctx: Context):
         """
         global moderator
 
-        document = await self.check_tmpc_channel(ctx, is_mod=True)
+        document = await self.check_tmpc_channel(ctx.author, ctx.channel.id, is_mod=True)
         await document.voice.set_permissions(moderator.item, overwrite=None)
         await document.chat.set_permissions(moderator.item, overwrite=None)
         embed: Embed = Embed(title="No Mod",
                              description=f"Mods can't see or join this channel anymore")
         await ctx.reply(embed=embed)
 
-    async def check_tmpc_channel(self, ctx: Context, is_mod: bool = False, everyone: bool = False) \
+    @tmpc.command(pass_context=True,
+                  brief="Invite a user to your temp channel",
+                  help="By default add the user to the study channel "
+                       "for gaming channel the first argument must be 'False'!\n"
+                       "Member must be invited by mentions.\n"
+                       "Limitations:\n"
+                       "- This command can be used 3/hour/user\n"
+                       "- The invite command is limited to 10 user at once")
+    @cooldown(3, 3600, BucketType.user)
+    async def invite(self, ctx: Context, study_channel: bool):
+
+        if study_channel:
+            channel = await self.study_db.find_one({DBKeyWrapperEnum.OWNER.value: ctx.author.id})
+        else:
+            channel = await self.gaming_db.find_one({DBKeyWrapperEnum.OWNER.value: ctx.author.id})
+        channel: Union[GamingChannel, StudyChannel]
+        document = await self.check_tmpc_channel(ctx.author, channel.chat.id)
+
+        voice_overwrites = document.voice.overwrites
+        chat_overwrites = document.chat.overwrites
+        msg: Message = ctx.message
+        if len(msg.mentions) > 10:
+            await ctx.reply("Warning hit limit")
+
+        changed = False
+        for mention in msg.mentions[:10]:
+            if mention not in chat_overwrites:
```

Comment:
maybe give a feedback to the user why a specific user couldn't be invited
So for example: "User#123 could not be invited because he either already is in the channel or already left by himself"

Reply:
You you're right.